Create .ckan files for Amazing Curve Editor and MechJebFARExt

BasharMilesTeg opened the pull request.

sarbian commented:
looks fine to me.

dbent commented on AmazingCurveEditor/AmazingCurveEditor-1.0.0.ckan:
```diff
@@ -0,0 +1,19 @@
+    "abstract": "Edit your Curves to Amaze your friends and family.",
+    "license": "GPL-2.0",
+    "author": "Sarbian",
+    "release_status": "stable",
```
Unless otherwise specified `stable` is the assumed value of `release_status`.

dbent commented on AmazingCurveEditor/AmazingCurveEditor-1.0.0.ckan:
```diff
@@ -0,0 +1,19 @@
+    "release_status": "stable",
+    "ksp_version": "1.0",
+    "resources": {
+        "x_ci": "https://ksp.sarbian.com/jenkins/job/AmazingCurveEditor/",
```
A [`ci` resource](https://github.com/KSP-CKAN/CKAN/blob/master/Spec.md#resources) exists since `v1.6` of the spec.

dbent commented on AmazingCurveEditor/AmazingCurveEditor-1.0.0.ckan:
```diff
@@ -0,0 +1,19 @@
+        "homepage": "http://forum.kerbalspaceprogram.com/threads/124708",
+        "repository": "https://github.com/sarbian/AmazingCurveEditor"
+    },
+    "version": "1.0.0",
```
The download itself has a version of `1.0.0.0`. `1.0.0` is not equivalent to `1.0.0.0` in version comparisons (`1.0.0.0` is greater).

dbent commented on AmazingCurveEditor/AmazingCurveEditor-1.0.0.ckan:
```diff
@@ -0,0 +1,19 @@
+    },
+    "version": "1.0.0",
+    "download": "https://ksp.sarbian.com/jenkins/job/AmazingCurveEditor/3/artifact/AmazingCurveEditor-1.0.0.0.zip",
+    "x_generated_by": "nyankan",
```
Eh?

dbent commented on AmazingCurveEditor/AmazingCurveEditor-1.0.0.ckan:
```diff
@@ -0,0 +1,19 @@
+    "version": "1.0.0",
+    "download": "https://ksp.sarbian.com/jenkins/job/AmazingCurveEditor/3/artifact/AmazingCurveEditor-1.0.0.0.zip",
+    "x_generated_by": "nyankan",
+    "download_size": 11851
```
It's not recommended to specify this unless the `.ckan` is generated by NetKAN.

dbent commented on AmazingCurveEditor/AmazingCurveEditor-1.0.0.ckan:
```diff
@@ -0,0 +1,19 @@
+    "license": "GPL-2.0",
+    "author": "Sarbian",
+    "release_status": "stable",
+    "ksp_version": "1.0",
```
Note, this will produce optimistic versioning and make the mod compatible with `1.0.ANYTHING`, including future releases. Sure you don't want?:
```json
{
    "ksp_version_min": "1.0.0",
    "ksp_version_max": "1.0.2"
}
```

dbent commented on MechJebFARExt/MechJebFARExt-2015-06-06.ckan:
```diff
@@ -0,0 +1,30 @@
+    "abstract": "Basic support for FAR inside MJ. For now it only adds surface control info inside MJ.",
+    "license": "GPL-3.0",
+    "author": "Sarbian",
+    "release_status": "stable",
```
Ditto as above.

dbent commented on MechJebFARExt/MechJebFARExt-2015-06-06.ckan:
```diff
@@ -0,0 +1,30 @@
+    "license": "GPL-3.0",
+    "author": "Sarbian",
+    "release_status": "stable",
+    "ksp_version": "1.0",
```
Ditto as above.

dbent commented on MechJebFARExt/MechJebFARExt-2015-06-06.ckan:
```diff
@@ -0,0 +1,30 @@
+        }
+    ],
+    "version": "2015-06-06",
+    "download": "https://www.sarbian.com/sarbian/MechJebFARExt.zip"
```
I assume this download URL isn't stable? i.e. it doesn't always produce the same version? That could be problematic once this updates as older `.ckan` files will be downloading the newest version.

BasharMilesTeg commented:
* I removed the "release_status" field for both files and updated the "spec_version" and "x_ci" fields for AmazingCurveEditor.
* I've used "version": "1.0.0" because Sarbian's mods traditionally use 3 digit version numbers in their release threads and ckan files even if the jenkins produced filenames use four digits.
* I use "x_generated_by": "nyankan" to indicate that I used netkan to generate the .ckan file and then clean it up by hand - remove "x_ci_version_base", remove the build number from "version" and change "download" to use the build number rather than 'lastStableBuild'. In addition that is why the "download_size" field is included.
* I set "ksp_version": "1.0" because the release thread specifies [1.0.x]. @sarbian would you like AmazingCurveEditor to only be listed for KSP versions 1.0.0, 1.0.1 and 1.0.2 or for any 1.0 version that Squad may release?
* Finally, @sarbian, would it be possible to create a copy of MechJebFARExt.zip with a version in the filename so that when you release another version it doesn't confuse the ckan clients caching?

dbent commented:
@BasharMilesTeg I see that `MechJebFARExt` has been updated with a permanent link, is this good to go now?
